.github/workflows/pull_request.yml

```diff
@@ -26,5 +26,17 @@ jobs:
                   cache: "yarn"
             - name: Install deps
               run: yarn
-            - name: Lint
+            - name: Test
               run: "yarn run test --coverage"
+            - name: Upload Artifact
+              uses: actions/upload-artifact@v3
+              with:
+                  name: coverage
+                  path: |
+                      coverage
+                      !coverage/lcov-report
+            - name: SonarCloud Scan
+              uses: SonarSource/sonarcloud-github-action@master
+              env:
+                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
```

Comment:
This won't work for forked PRs, thus will mean you can't use coverage as a branch protection

Reply:
https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

Reply:
Makes sense! I'm happy to punt that for now as we don't expect contributions for now.
But I have tracked it as part of https://github.com/vector-im/compound/issues/80